### PR TITLE
Queue

### PR DESCRIPTION
**Pull request by vbenavente**

### CODE REVIEW

File: src/queue.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    def __init__(self, iterable=None):
+        self.dll = dll.DoublyLinkedList(iterable)
+
+    def enqueue(self, data):
+        """Add a value to the head of the queue."""
+        return self.dll.push(data)
+
+    def dequeue(self):
+        """Remove a value from the tail of the queue."""
+        return self.dll.remove(self.dll.tail.data)
```
Comment (WillWeatherford) [inline src/queue.py]: remove() is not the appropriate DLL method to be calling on here. Remove is an expensive loop through the whole list.

Also, if you are saying to remove the tail's data in order to dequeue, what happens when your list has two of that same value in it? e.g.
`Queue([1,2,3,2])`
What would happen when you call dequeue on this instance?

File: src/queue.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    def peek(self):
+        """Return value of tail's previous node w/o removing it."""
+        if self.dll.tail is None:
+            return None
+        return self.dll.tail._prev.data
```
Comment (WillWeatherford) [inline src/queue.py]: This definitely is not right and is causing some of my tests to fail with attribute error. This reveals that your tests need to include instances where you push only one item into the Queue.

File: src/queue.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+    def size(self):
+        """Return size of the queue, 0 if empty."""
+        if self.dll.head is None:
+            return 0
```
Comment (WillWeatherford) [inline src/queue.py]: FYI, lines 29 and 30 are unnecessary. I won't take a grade off for this, but see if you can identify why.

File: src/test_queue.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+"""File tests our queue data structure."""
+from __future__ import unicode_literals
+from queue import Queue
+
+TEST_QUEUE_DATA = [
+    [1, 2, 3, 4, 5],
+    ("cup", "coffee", "pastry")
+]
```
Comment (WillWeatherford) [inline src/test_queue.py]: This is an insufficient diversity of test cases. Here you're checking only checking a single input of length 2. What about inputs of size 0 or size 1? What about size 1000? How large of a range can your Queue handle?

File: src/test_queue.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    """Test that enqueue'd data is added to the head of the queue."""
+    test_queue = Queue(TEST_QUEUE_DATA)
+    test_queue.enqueue("stripes")
+    assert test_queue.dll.head.data == "stripes"
+
+
+def test_queue_dequeue():
+    """Test dequeue'd node is no longer in queue."""
+    test_queue = Queue(TEST_QUEUE_DATA)
+    assert test_queue.dequeue() == [1, 2, 3, 4, 5]
```
Comment (WillWeatherford) [inline src/test_queue.py]: You need also to test dequeue on an empty Queue.
